src/ui/server/helpers/summary-lists/your-buyer/working-with-buyer/index.ts

```diff
@@ -41,6 +43,41 @@ const workingWithBuyerFields = (answers: ApplicationBuyer, referenceNumber: numb
       },
       mapYesNoField(answers[CONNECTION_WITH_BUYER]),
     ),
+  ] as Array<SummaryListItemData>;
+
+  /**
+   * if CONNECTION_WITH_BUYER is true
+   * pushes CONNECTION_WITH_BUYER_DESCRIPTION to the summary list
+   */
+  if (answers[CONNECTION_WITH_BUYER]) {
+    fields.push(
+      fieldGroupItem({
+        field: getFieldById(FIELDS, CONNECTION_WITH_BUYER_DESCRIPTION),
+        data: answers,
+        href: generateChangeLink(
+          CONNECTION_WITH_BUYER_CHANGE,
+          CONNECTION_WITH_BUYER_CHECK_AND_CHANGE,
+          `#${CONNECTION_WITH_BUYER_DESCRIPTION}-label`,
+          referenceNumber,
+          checkAndChange,
+        ),
+        renderChangeLink: true,
+      }),
+    );
+  }
+
+  return fields;
+};
+/**
+ * workingWithBuyerFields
+ * Create all working with buyer fields and values for the Insurance - your buyer govukSummaryList
+ * @param {ApplicationBuyer} answers buyer data
+ * @param {Number} referenceNumber application reference number
+ * @returns {Object} All working with buyer fields and values in an object structure for GOVUK summary list structure
+ */
```

Comment:
Missing `checkAndChange` docuemntation, also I think the `returns` is invalid, should be array?